pkg/model/mselection/heaviest.go

```diff
@@ -328,22 +303,62 @@ func (s *HeaviestSelector) removeTip(it *item) {
 	it.tip = nil
 }
 
-// tipItems returns a copy of the items corresponding to tips.
-func (s *HeaviestSelector) tipItems() []*item {
+// copyTipItemsToList returns a copy of the items corresponding to tips.
+func (s *HeaviestSelector) copyTipItemsToList() *itemList {
 	s.Lock()
 	defer s.Unlock()
 
 	result := make([]*item, 0, s.tips.Len())
 	for e := s.tips.Front(); e != nil; e = e.Next() {
 		result = append(result, e.Value.(*item))
 	}
-	return result
+	return &itemList{items: result}
+}
+
+// GetStats returns the amount of known tips and approvees of s.
+func (s *HeaviestSelector) GetStats() (tipCount int, approveeCount int) {
+	return s.tips.Len(), len(s.approvers)
+}
+
+// checkBelowMaxDepth checks the below max depth criteria for the given approvees.
+// if one of the approvees fails the below max depth criteria, the result is false.
+func checkBelowMaxDepth(approveeHashes map[string]struct{}) bool {
```

Comment:
This check should be implemented as part of URTS as well; can we use one implementation only?